xpsparser/parser.py

```diff
@@ -0,0 +1,112 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# import datetime
+# import numpy as np
+import json
+import numpy as np
+from datetime import datetime
+import logging
+from nomad.units import ureg
+
+from nomad.datamodel import EntryArchive
+from nomad.parsing import FairdiParser
+
+from . import metainfo  # pylint: disable=unused-import
+from .metainfo import *
+'''
+This is a test parser for XPS Parser
+'''
+
+logger = logging.getLogger(__name__)
+
+
+class XPSParser(FairdiParser):
+    def __init__(self):
+        super().__init__(
```

Comment:
This will not be able to distinguish between the different JSON flavours and parsers. The first parser in the parsers list that accepts JSON will be used. Now that we have different JSON parsers, we have to adapt the matching. I can start thinking about something JSON specific. In the meantime a regular expression on the filename might suffice (e.g. `.*\.xps.json` vs `.*\.eels.json` or something like this). You can use `mainfile_name_re` to specify that.